metron-stellar/stellar-zeppelin/src/main/java/org/apache/metron/stellar/zeppelin/StellarInterpreter.java

```diff
@@ -0,0 +1,158 @@
+   *
+   * See https://zeppelin.apache.org/docs/latest/manual/interpreters.html#interpreter-binding-mode.
+   */
+  private StellarShellExecutor executor;
+
+  /**
+   * Handles auto-completion for Stellar expressions.
+   */
+  private StellarAutoCompleter autoCompleter;
+
+  public StellarInterpreter(Properties properties) {
+    super(properties);
+    this.autoCompleter = new DefaultStellarAutoCompleter();
+  }
+
+  public void open() {
+    try {
+      executor = createExecutor();
+
+    } catch (Exception e) {
+      LOG.error("Unable to create a StellarShellExecutor", e);
+    }
+  }
+
+  public void close() {
+    // nothing to do
+  }
+
+  public StellarShellExecutor createExecutor() throws Exception {
+
+    Properties props = getProperty();
+    StellarShellExecutor executor = new DefaultStellarShellExecutor(props, Optional.empty());
+
+    // register the auto-completer to be notified
+    executor.addSpecialListener((magic) -> autoCompleter.addCandidateFunction(magic.getCommand()));
+    executor.addFunctionListener((fn) -> autoCompleter.addCandidateFunction(fn.getName()));
+    executor.addVariableListener((name, val) -> autoCompleter.addCandidateVariable(name));
+
+    executor.init();
+    return executor;
+  }
+
+  public InterpreterResult interpret(String input, InterpreterContext context) {
+    InterpreterResult result;
+
+    try {
+      // execute the input
+      StellarResult stellarResult = executor.execute(input);
+
+      if(stellarResult.isSuccess()) {
+        // on success - if no result, use a blank value
+        Object value = stellarResult.getValue().orElse("");
+        String text = ConversionUtils.convert(value, String.class);
+        result = new InterpreterResult(SUCCESS, TEXT, text);
+
+      } else if(stellarResult.isError()) {
+        // on error
+        Throwable e = stellarResult.getException().get();
+        String error = ExceptionUtils.getRootCauseMessage(e);
+        String stack = ExceptionUtils.getStackTrace(e);
+        result = new InterpreterResult(ERROR, TEXT, error + System.lineSeparator() + stack);
+
+      } else {
+        // should not happen
+        throw new IllegalStateException("Unexpected Stellar result status. Please file a bug report.");
+      }
+
+    } catch(Throwable t) {
+      // unexpected exception
+      String message = ExceptionUtils.getRootCauseMessage(t);
+      String stack = ExceptionUtils.getStackTrace(t);
+      result = new InterpreterResult(ERROR, TEXT, message + System.lineSeparator() + stack);
+    }
+
+    return result;
+  }
+
+  public void cancel(InterpreterContext context) {
+    // there is no way to cancel the execution of a Stellar expression
+  }
+
+  public FormType getFormType() {
+    return FormType.SIMPLE;
+  }
+
+  public int getProgress(InterpreterContext context) {
+    // unable to provide progress
+    return 0;
+  }
+
+  @Override
+  public List<InterpreterCompletion> completion(String buf, int cursor) {
```

Comment:
This is where we connect our generic auto-completer that is used across both REPLs into the auto-complete extension point offered by Zeppelin.